src/support/json.h

```diff
@@ -272,12 +272,17 @@ struct Value {
       // String
       // Start |close| at the opening ", and in the loop below we will always
       // begin looking at the first character after.
-      char* close = curr;
-      // Skip escaped "
-      do {
-        close = strchr(close + 1, '"');
-      } while (*(close - 1) == '\\');
-      assert(close);
+      char* close = curr + 1;
+      // Skip escaped ", which appears as \". We need to be careful though, as
+      // \" might also be \\" which would be an escaped \ and an *un*escaped ".
+      while (*close && *close != '"') {
+        if (*close == '\\') {
+          // Skip the \ and the character after it, which it escapes.
+          close++;
+          assert(*close);
```

Comment:
Not sure how useful this assertion is.

Reply:
It is necessary, because if this is 0, then it is not valid to do the next `close++` and then read `*close`, which would be past the string.

Reply:
Oh, I see, it's checking for the end of a C-style string. Can we add a `&& "unexpected end of string"` to clarify?